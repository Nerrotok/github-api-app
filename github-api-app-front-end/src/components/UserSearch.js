import React from "react";

class UserSearch extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      searchInputValue: "",
    };

    // Bind event handlers
    this.formSubmissionHandler = this.formSubmissionHandler.bind(this);
    this.inputChangeHandler = this.inputChangeHandler.bind(this);
  }

  // handle input change
  inputChangeHandler(e) {
    this.setState({ searchInputValue: e.target.value });
  }

  // handle form submission
  formSubmissionHandler(e) {
    e.preventDefault();

    // get state and parent functions
    const { searchInputValue } = this.state;
    const { setUserName, setUserSelectedStatus, setRepoSelectedStatus } =
      this.props;

    // reset userSelectedStatus and selectedRepoStatus
    setUserSelectedStatus(false);
    setRepoSelectedStatus(false);

    // Update parent component
    setUserName(searchInputValue);

    // submit POST request
    fetch("http://localhost:8080/api/user", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        accountUserName: searchInputValue,
      }),
    })
      .then((res) => res.json())
      .then(
        (result) => {
          if (result.message === "Internal Server Error") {
            alert(`Error: ${result.message}`);
          } else {
            setUserSelectedStatus(true);
          }
        },
        (error) => {
          alert("Error occured: " + error.message);
        }
      );
  }

  render() {
    return (
      <div className="UserSearch--container">
        <h3 className="UserSearch--subheading">Search User:</h3>
        <form
          onSubmit={this.formSubmissionHandler}
          className="UserSearch--form"
        >
          <label className="UserSearch--label">User name:</label>
          <input
            onChange={this.inputChangeHandler}
            className="UserSearch--input"
            type="text"
            value={this.state.searchInputValue}
            required
          />
          <button className="UserSearch--button" type="submit">
            Search
          </button>
        </form>
      </div>
    );
  }
}

export default UserSearch;
