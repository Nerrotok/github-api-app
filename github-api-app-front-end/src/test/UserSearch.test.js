import React from "react";
import UserSearch from "../components/UserSearch";
import renderer from "react-test-renderer";

test("UserSearch renders correctly", () => {
  const tree = renderer.create(<UserSearch />).toJSON();
  expect(tree).toMatchSnapshot();
});
