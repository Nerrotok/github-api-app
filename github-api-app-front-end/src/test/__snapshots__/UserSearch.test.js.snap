// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="UserSearch--container"
>
  <h3
    className="UserSearch--subheading"
  >
    Search User:
  </h3>
  <form
    className="UserSearch--form"
    onSubmit={[Function]}
  >
    <label
      className="UserSearch--label"
    >
      User name:
    </label>
    <input
      className="UserSearch--input"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <button
      className="UserSearch--button"
      type="submit"
    >
      Search
    </button>
  </form>
</div>
`;

exports[`UserSearch renders correctly 1`] = `
<div
  className="UserSearch--container"
>
  <h3
    className="UserSearch--subheading"
  >
    Search User:
  </h3>
  <form
    className="UserSearch--form"
    onSubmit={[Function]}
  >
    <label
      className="UserSearch--label"
    >
      User name:
    </label>
    <input
      className="UserSearch--input"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <button
      className="UserSearch--button"
      type="submit"
    >
      Search
    </button>
  </form>
</div>
`;
